devU-api/src/containerAutoGrader/test/containerAutoGrade.controller.test.ts

```diff
@@ -0,0 +1,224 @@
+import { UpdateResult } from "typeorm"
+
+import { ContainerAutoGrader } from '../../../../devu-shared/src'
```

Comment:
Does this work when running the app in docker-compose? It shouldn't because the dockerfile is not copying shared into the container. Did you only test this locally?

The fix is to add COPY shared into the Dockerfiles for api and client so we can finally fix this mess

Reply:
this was what I mentioned in Discord. should we make change with the dockerfile in devU-api which moves docekrfile to the devU folder so that the import from "../../../../devu-shared/src" could stay as it is, or create another container for devu-shared module but need to modify it when testing locally and running in docker-compose.

I have pushed one commit before, creating a "Dockerfile-api" file in the devU folder and letting it handle copying shared modules into web container. But I rollback the commit because Nicholas said this needs discussion before making this change.

Reply:
Yes, we'll have to discus this. Not sure what the best approach is yet